Create VS Code extension post (#5116)
sourcegraph/about: blogposts/2022/sourcegraph-extension-for-vs-code.md

author: Murat Sutunc
publishDate: 2022-02-08T00:00-07:00
tags: [blog]
slug: ways-to-use-sourcegraph-extension-for-vs-code
heroImage: https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png
socialImage: https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png
published: true
---

![Explore millions of open source repositories right from your IDE](https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png)

My name is [Murat Sutunc](https://handbook.sourcegraph.com/team#murat-sutunc), and I’m the Engineering Manager for the Growth and Integrations team at Sourcegraph. Our team currently owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our IDE and browser extensions), and Sourcegraph extensions. As someone who does a lot of code reviews and occasionally jumps in to fix small issues, I wanted to share with everyone how I use my favorite IDE (VS Code!) and Sourcegraph together.

> TL;DR 👉 The Sourcegraph VS Code extension is a really powerful tool that lets you search your code and 2M+ open source repositories, all from your IDE! It’s available on [the marketplace](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph)—no account needed, give it a try!

Author: Wondering if we should give this some kind of different style treatment, e.g. like at the top of this post: https://about.sourcegraph.com/blog/log4j-log4shell-0-day/?

@muratsu @iskyOS

Reviewer: @rebeccadee I like calling out that section! We could definitely use that alert box style.

### Finding examples to fix issues

> Now and then, I jump in to help our team with some issues that are not on the critical path. For those unfamiliar, I think it's fair to say Sourcegraph's codebase is sizeable. It still fits on my laptop hard drive, but it’s large enough to tire IDEs when indexing. On top of our large main repository, we have several hundred additional repositories, and quickly finding what I'm looking for can be challenging.

Author: @muratsu Could we link to our codebase to demonstrate what you mean by sizeable? I'm sure people will be curious as you say it!

Reviewer: ```suggestion
Now and then, I jump in to help our team with some issues that are not on the critical path. For those unfamiliar, I think it's fair to say [Sourcegraph's codebase is sizeable](https://github.com/sourcegraph/sourcegraph). It still fits on my laptop hard drive, but it’s large enough to tire IDEs when indexing. On top of our large main repository, we have several hundred additional repositories, and quickly finding what I'm looking for can be challenging.
```

Reviewer 2: Can we link to https://github.com/sourcegraph instead? I think the fact that we have 530 repositories is also important. Not only a med size primary codebase but also bits of code sits in multiple repos.

When working on a plugin issue, I first start with the repository that the issue is filed against. Once I familiarize myself with the issue, I tend to fire up VS Code to work on my solution. Frequently I run into cases where I remember solving a similar issue before but cannot remember how I solved it at the time. One such example is when I needed to write some code involving the [rxjs](https://rxjs.dev/) library but forgot the exact syntax because I’m not very familiar with the library.

I ran the following search query to find all Sourcegraph extensions we have (repository name pattern is sourcegraph-extension-name) which make use of the rxjs library:

```
repo:^github\.com/sourcegraph/sourcegraph-.+$  /import .+ from 'rxjs'/
```

<video loop autoplay muted playsinline>
  <source src="https://sourcegraphstatic.com/vs-code-extension/vs-code-regex-usecase.webm" type="video/webm" />
  <source src="https://sourcegraphstatic.com//vs-code-extension/vs-code-regex-usecase.mp4" type="video/mp4" />

> </video>

Reviewer: @rebeccadee It looks like no files exist at those URLs — I can try moving them to GCP

> </video>

Author: @iskyOS I did upload them there but I may have used the wrong URL structure 🤦‍♀️ Let me see what I can do. Thanks!

> </video>

Reviewer: Oh sorry missed your later message! OK, sounds good.

Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome.

> ### Researching an error

Author: @iskyOS I wonder if we should retitle this section to align better to the use case? 
```suggestion
### Researching and fixing vulnerabilities
```

published: true
---

![Explore millions of open source repositories right from your IDE](https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png)

My name is [Murat Sutunc](https://handbook.sourcegraph.com/team#murat-sutunc), and I’m the Engineering Manager for the Growth and Integrations team at Sourcegraph. Our team currently owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our IDE and browser extensions), and Sourcegraph extensions. As someone who does a lot of code reviews and occasionally jumps in to fix small issues, I wanted to share with everyone how I use my favorite IDE (VS Code!) and Sourcegraph together.

<div class="alert alert-primary" role="alert" style="max-width:650px; align-items: center; margin: auto auto 20px auto">
      <span>TL;DR 👉 The Sourcegraph VS Code extension is a really powerful tool that lets you search your code and 2M+ open source repositories, all from your IDE! It’s available on <a = href="https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph">the marketplace</a>—no account needed, give it a try!</span>
</div>

> ### Finding examples to fix issues

Reviewer: ```suggestion
### Finding reusable code to fix issues
```

When working on a plugin issue, I first start with the repository that the issue is filed against. Once I familiarize myself with the issue, I tend to fire up VS Code to work on my solution. Frequently I run into cases where I remember solving a similar issue before but cannot remember how I solved it at the time. One such example is when I needed to write some code involving the [rxjs](https://rxjs.dev/) library but forgot the exact syntax because I’m not very familiar with the library.

I ran the following search query to find all Sourcegraph extensions we have (repository name pattern is sourcegraph-extension-name) which make use of the rxjs library:

```
repo:^github\.com/sourcegraph/sourcegraph-.+$  /import .+ from 'rxjs'/
```

<video loop autoplay muted playsinline>
  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-extension/vs-code-regex-usecase.webm" type="video/webm" />
  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-extension/vs-code-regex-usecase.mp4" type="video/mp4" />
</video>

> Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome.

Reviewer: This sentence reads a little funny to me with present/past tense — @rebeccadee is this grammatically correct?

Reviewer: ```suggestion
Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome. I can quickly reuse that code to solve my issue without reinventing the wheel, too.
```

Reviewer: (Adding some clarity that this is about code resuse)

```
gradle org\.apache\.logging\.log4j ['"] 2\.(0|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16)(\.[0-9]+)
```

As you can see in the video below, the global search returns several repositories that require further investigation. I finally changed the search context to Sourcegraph to narrow down my search scope to find all the repositories affecting our organization.

<video loop autoplay muted playsinline>
  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.webm" type="video/webm" />
  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.mp4" type="video/mp4" />
</video>

> I hope that you find these workflows useful. If you have questions or suggestions, you can reach out to me on Twitter ([@muratsutunc](http://twitter.com/muratsutunc)). You can [download the Sourcegraph VS Code extension from this link](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph). Happy hacking!

Reviewer: ```suggestion
I hope that you find these workflows useful. The extension has become a handy part of my day-to-day to stay in flow without leaving my IDE, especially for finding code to reuse or to quickly find and resolve vulnerabilities. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.

If you have questions or suggestions, you can reach out to me on Twitter ([@muratsutunc](http://twitter.com/muratsutunc)). You can [download the Sourcegraph VS Code extension from this link](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph). Happy hacking!
```

---
title: 'How I use the Sourcegraph extension for VS Code'

> description: 'Sourcegraph’s extension for Visual Studio Code allows you to search millions of open source repositories without cloning them to your local machine or leaving your IDE. Here’s how one engineer integrates it into his workflow.'

Author: ```suggestion
description: 'Sourcegraph’s extension for Visual Studio Code allows you to search millions of open source repositories without cloning them to your local machine or leaving your IDE. Here’s how one engineering manager integrates it into his workflow.'
```

```
gradle org\.apache\.logging\.log4j ['"] 2\.(0|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16)(\.[0-9]+)
```

As you can see in the video below, the global search returns several repositories that require further investigation. I finally changed the search context to Sourcegraph to narrow down my search scope to find all the repositories affecting our organization.

<video loop autoplay muted playsinline>
  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.webm" type="video/webm" />
  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.mp4" type="video/mp4" />
</video>

> I hope that you find these workflows useful. The extension has become a handy part of my day-to-day to stay in flow without leaving my IDE, especially for finding code to reuse or to quickly find and resolve vulnerabilities. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.

Author: ```suggestion
I hope you find these workflows useful. The extension has become a handy part of my day-to-day flow, especially for finding code to reuse or to quickly find and resolve vulnerabilities without leaving my IDE. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.
```
